### PR TITLE
feat: Add support for aws_ecr_repository_creation_template

### PR DESCRIPTION
**Pull request by jakubmatyszewski**
## Description
Add support for aws_ecr_repository_creation_template

## Motivation and Context
aws_ecr_repository_creation_template was added in terraform aws provider [5.61](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.61.0).

## Breaking Changes
- no

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -172,6 +172,11 @@ module "ecr_registry" {
     }]
   }]
 
+  # Registry Replication Configuration
```
Comment (antonbabenko) [inline examples/complete/main.tf:175]:
```suggestion
  # Registry Creation Template
```

File: variables.tf [hunk 2]
```diff
@@ -197,3 +197,67 @@ variable "registry_replication_rules" {
...
+variable "repository_creation_template_image_tag_mutability" {
+  description = "The tag mutability setting for any created repositories. Must be one of: MUTABLE or IMMUTABLE. Defaults to MUTABLE."
+  type        = string
+  default     = "MUTABLE"
+}
+
+variable "repository_creation_template_lifecycle_policy" {
+  description = "The lifecycle policy document to apply to any created repositories. See more details about Policy Parameters in the official AWS docs. Consider using the aws_ecr_lifecycle_policy_document data_source to generate/manage the JSON document used for the lifecycle_policy argument."
+  type        = string
+  default     = ""
```
Comment (antonbabenko) [inline variables.tf:244]:
```suggestion
  default     = null
```

File: variables.tf [hunk 3]
```diff
@@ -197,3 +197,67 @@ variable "registry_replication_rules" {
...
+variable "repository_creation_template_custom_role_arn" {
+  description = "A custom IAM role to use for repository creation. Required if using repository tags or KMS encryption."
+  type        = string
+  default     = ""
+}
+
+variable "repository_creation_template_description" {
+  description = "The description for this template."
+  type        = string
+  default     = ""
```
Comment (antonbabenko) [inline variables.tf:232]:
```suggestion
  default     = null
```

File: variables.tf [hunk 4]
```diff
@@ -197,3 +197,67 @@ variable "registry_replication_rules" {
...
+variable "repository_creation_template_applied_for" {
+  description = "Which features this template applies to. Must contain one or more of PULL_THROUGH_CACHE or REPLICATION."
+  type        = list(string)
+  default     = ["PULL_THROUGH_CACHE"]
+}
+
+variable "repository_creation_template_custom_role_arn" {
+  description = "A custom IAM role to use for repository creation. Required if using repository tags or KMS encryption."
+  type        = string
+  default     = ""
```
Comment (antonbabenko) [inline variables.tf:226]:
```suggestion
  default     = null
```

**Review by antonbabenko — CHANGES_REQUESTED** [review]

**Review by bryantbiggs — CHANGES_REQUESTED** [review]
just putting a hold on this since theres quite a bit we'll need to change - most likely this resource will live in its own sub-module

the ECR team has asked me to take a look - will get to this next week

**Comment by amontalban** [comment]
Hey @bryantbiggs, just checking in to see if you are working on this feature because we are about to set multiple ECR repositories and will benefit from this. Thanks in advance!

**Comment by bryantbiggs** [comment]
working on it this morning as we speak - should know more about rough ETA later today

**Comment by bryantbiggs** [comment]
need to revisit tomorrow with fresh eyes but you can take a look here - happy for any early feedback https://github.com/terraform-aws-modules/terraform-aws-ecr/pull/46

**Comment by jakubmatyszewski** [comment]
closing this PR, thanks @bryantbiggs for taking over

**Pull request closed without merging**